feeds/wifi-trunk/hostapd/patches/908-ubus-fix-client-events.patch

```diff
@@ -0,0 +1,111 @@
+--- a/src/ap/ubus.c
++++ b/src/ap/ubus.c
+@@ -578,6 +578,7 @@ static int hostapd_clear_session(struct
+ 	{
+ 		if (session_id == rec->session_id) {
+ 			/* dec counter and delete */
++			wpa_printf(MSG_INFO, "%s Deleting client session with session id %llu", __func__, session_id);
+ 			cached_events_nr -= rec->rec_nr;
+ 			avl_delete(&hapd->ubus.rt_events, &rec->avl);
+ 			os_free(rec->records);
+@@ -631,6 +632,10 @@ static int hostapd_sessions(struct ubus_
+ 		/* messages for current session */
+ 		for (size_t i = 0; i < rec->rec_nr; i++) {
+ 			c_rec = &rec->records[i];
++			if (c_rec->processed) {
++				/* Record is already reported, continue */
++				continue;
++			}
+ 			/* check message type */
+ 			switch (c_rec->type) {
+ 			/* ClientAuthEvent */
+@@ -644,6 +649,7 @@ static int hostapd_sessions(struct ubus_
+ 				blobmsg_add_u32(&b_ev, "auth_status", p->auth_status);
+ 				blobmsg_add_string(&b_ev, "ssid", p->ssid);
+ 				blobmsg_close_table(&b_ev, t_msg);
++				c_rec->processed = true;
```

Comment:
Just a thought. why don't we just delete the records instead of marking them as processed.

Reply:
These are individual events which are part of a client session and  client session as a whole is cleared by SM when it has processed all the events.